infra/Pos/Recovery/Info.hs

```diff
@@ -2,24 +2,88 @@
 -- and provide helpers related to it.
 
 module Pos.Recovery.Info
-       ( MonadRecoveryInfo(..)
+       ( SyncStatus (..)
+       , MonadRecoveryInfo(..)
+       , recoveryInProgress
        , recoveryCommGuard
        ) where
 
-import           System.Wlog (WithLogger, logDebug)
 import           Universum
 
+import qualified Data.Text.Buildable
+import           Formatting          (bprint, build, sformat, stext, (%))
+import           System.Wlog         (WithLogger, logDebug)
+
+import           Pos.Core            (HasCoreConstants, SlotCount, SlotId, slotIdF,
+                                      slotSecurityParam)
+
+-- | An algebraic data type which represents how well we are
+-- synchronized with the network.
+data SyncStatus
+    = SSDoingRecovery
+    -- ^ We are doing recovery right now, so it's better to wait until
+    -- we finish.
+    | SSUnknownSlot
+    -- ^ We don't know current slot, so we are definitely not
+    -- synchronized well enough.
+    | SSLagBehind { sslbTipSlot     :: !SlotId
+                  -- ^ Our tip's slot (if our tip is genesis, we use 0
+                  -- as local slot).
+                  , sslbCurrentSlot :: !SlotId }
+    -- ^ We know current slot, but our tip's slot lags behind current
```

Comment:
AFAIU in any case of lag, our tip belong to current epoch. is it really `too much`?

Reply:
In this particular type meaning of `too much` is left flexible. The user of this type ought to define how much it `too much` for them (and pass this value to `getSyncStatus`). So too much here means `more than a threshold specified by user of this type`.
If you are talking about concrete value (`slotSecurityParam`), then I think it's subjective and probably should be different for different workers (but that was too much for this task).
Anyway, the fact that our tip belongs to current epoch doesn't mean anything, we may lag behind for 4.5 days and in this case there are not many useful things we can do. And anyway, if we are lagging behind for more than `2k` blocks, we are going recovery or will start doing it soon.